server/src/test/java/org/elasticsearch/cluster/health/ClusterIndexHealthTests.java

```diff
@@ -141,187 +141,47 @@ protected Predicate<String> getRandomFieldsExcludeFilter() {
-                    status,
-                    instance.getShards()
-                );
-            case "shards":
-                Map<Integer, ClusterShardHealth> map;
-                if (instance.getShards().isEmpty()) {
-                    map = Collections.singletonMap(0, ClusterShardHealthTests.randomShardHealth(0));
-                } else {
-                    map = new HashMap<>(instance.getShards());
-                    map.remove(map.keySet().iterator().next());
-                }
-                return new ClusterIndexHealth(
-                    instance.getIndex(),
-                    instance.getNumberOfShards(),
-                    instance.getNumberOfReplicas(),
-                    instance.getActiveShards(),
-                    instance.getRelocatingShards(),
-                    instance.getInitializingShards(),
-                    instance.getUnassignedShards(),
-                    instance.getUnassignedPrimaryShards(),
-                    instance.getActivePrimaryShards(),
-                    instance.getStatus(),
-                    map
-                );
-            default:
-                throw new UnsupportedOperationException();
+        String index = instance.getIndex();
+        int numberOfShards = instance.getNumberOfShards();
+        int numberOfReplicas = instance.getNumberOfReplicas();
+        int activeShards = instance.getActiveShards();
+        int relocatingShards = instance.getRelocatingShards();
+        int initializingShards = instance.getInitializingShards();
+        int unassignedShards = instance.getUnassignedShards();
+        int unassignedPrimaryShards = instance.getUnassignedPrimaryShards();
+        int activePrimaryShards = instance.getActivePrimaryShards();
+        ClusterHealthStatus status = instance.getStatus();
+        Map<Integer, ClusterShardHealth> shards = instance.getShards();
+
+        switch (randomIntBetween(0, 9)) {
+            case 0 -> index = instance.getIndex() + randomAlphaOfLengthBetween(2, 5);
+            case 1 -> numberOfShards = instance.getNumberOfShards() + between(1, 10);
+            case 2 -> numberOfReplicas = instance.getNumberOfReplicas() + between(1, 10);
+            case 3 -> activeShards = instance.getActiveShards() + between(1, 10);
+            case 4 -> relocatingShards = instance.getRelocatingShards() + between(1, 10);
+            case 5 -> initializingShards = instance.getInitializingShards() + between(1, 10);
+            case 6 -> unassignedShards = instance.getUnassignedShards() + between(1, 10);
+            case 7 -> unassignedPrimaryShards = instance.getUnassignedPrimaryShards() + between(1, 10);
+            case 8 -> activePrimaryShards = instance.getActivePrimaryShards() + between(1, 10);
+            case 9 -> status = randomFrom(
+                Arrays.stream(ClusterHealthStatus.values()).filter(value -> value != instance.getStatus()).toList()
+            );
```

Comment:
Tiny (optional) nit:
```suggestion
            case 9 -> status = randomValueOtherThan(instance.getStatus(), () -> randomFrom(ClusterHealthStatus.values()));
```
This is just a personal preference, so feel free to stick to the original code.

Reply:
I like that style better :-) Switching it over